src/core/modules/SolarSystem.cpp

```diff
@@ -1059,12 +1066,27 @@ bool SolarSystem::loadPlanets(const QString& filePath)
 			QSharedPointer<Comet> mp = newP.dynamicCast<Comet>();
 
 			//g,k magnitude system
-			const float magnitude = pd.value(secname+"/absolute_magnitude", -99).toFloat();
+			const float magnitude = pd.value(secname+"/absolute_magnitude", -99.f).toFloat();
 			const float slope = qBound(-5.0f, pd.value(secname+"/slope_parameter", 4.0f).toFloat(), 30.0f);
-			if (magnitude > -99)
-			{
+			if (magnitude > -99.f)
 					mp->setAbsoluteMagnitudeAndSlope(magnitude, slope);
-			}
+
+			QString iauDesignation = pd.value(secname+"/iau_designation", "").toString();
+			if (!iauDesignation.isEmpty())
+				mp->setIAUDesignation(iauDesignation);
+			// order of codes: date_code [P/1982 U1] - perihelion_code [1986 III] - discovery_code [1982i]
+			QStringList codes = { pd.value(secname+"/date_code", "").toString(),
```

Comment:
In light of the recent updates, it seems date_code is now redundant (?). iau_designation looks like P/1982 U1, same as date_code. 

The only (minor) issue now is the collision of "P/1982 U1" and "name=1P/Halley". I am unsure whether this would better be written "1P/1982 U1 Halley". However, "1P/Halley (P/1982 U1 -  1982i - 1986III)" is perfectly understandable.

Reply:
`date_code` will be empty almost always, but in some cases the supporting this parameter will be important for periodical comets with standard designations XXP.

Reply:
That means, the complete infoString label according to MPC standards and our aim to protect existing labels should be "1P/1682 U1 Halley (P/1982 U1 - 1982i - 1986III)"? And the screen label "1P/1682 U1 Halley"?
 
Ouch, this reads so wrong! It would make life easier to read the label as "Halley in its apparition that started in 1982". 

But I agree, we must show the standardized name first.

Reply:
No, `1P/Halley (P/1982 U1 - 1982i - 1986III)`

Reply:
Hmm… Kronk use `1P/1682 Q1` style, MPC use `P/1682 Q1` style

Reply:
https://pds-smallbodies.astro.umd.edu/data_sb/resources/designation_formats.shtml